chat-server/src/routes/conversations/addMessageToConversation.ts

```diff
@@ -103,7 +111,7 @@ export function makeAddMessageToConversationRoute({
       if (!conversationInDb) {
         return sendErrorResponse(res, 404, "Conversation not found");
       }
-      if (conversationInDb.ipAddress !== ipAddress) {
+      if (!areEquivalentIpAddresses(conversationInDb.ipAddress, ip)) {
```

Comment:
We have essentially this same block in all three routes. Could we abstract to middleware?

Reply:
interesting idea. doing so would require an extra DB round trip per request. unless we were to add the conversation to the request in the middleware, which would seem a little *funky* to me. it'd also require the DB request to get conversation even in scenarios where the req would otherwise fail before DB req necessary. 

i'm open to adding to request in middleware, though think that could make the code a bit more confusing even if reducing a bit of boilerplate.

curious your thoughts here.

Reply:
Reasonable enough to keep it like this for now. Maybe if/when we add more routes we can put in a more shareable abstraction.